### PR TITLE
PIM-4613 : fix sorted dates features

### PR DESCRIPTION
**Pull request by jmleroux**
| Q                    | A
| -------------------- | ---
| Bug fix?             |y
| New feature?         |n
| BC breaks?           |n
| CI currently passes? |y
| Tests pass?          |y
| Scenarios pass?      |y
| Checkstyle issues?*  |n
| PMD issues?**        |n
| Changelog updated?   |n
| Fixed tickets        |PIM-4613
| DB schema updated?   |
| Migration script?    |
| Doc PR               |

### CODE REVIEW

**Comment by solivier** [comment]
GTM

File: features/Context/Page/Base/Grid.php [hunk 1]
```diff
@@ -389,7 +389,18 @@ public function isSortedAndOrdered($column, $order)
             return false;
         }
 
-        $values       = $this->getValuesInColumn($column);
+        $values = $this->getValuesInColumn($column);
+
+        // format date values before sorting
```
Comment (BitOne) [inline features/Context/Page/Base/Grid.php]: If you feel the need to put this kind of comment, it certainly means that the core following it should be put in a specific function ;)

File: features/Context/Page/Base/Grid.php [hunk 2]
```diff
@@ -187,7 +187,7 @@ public function filterBy($filterName, $value, $operator = false, DriverInterface
 
             if ($operator !== false) {
```
Comment (phaseinducer) [inline features/Context/Page/Base/Grid.php]: yoda style

File: features/Context/Page/Base/Grid.php [hunk 3]
```diff
@@ -187,7 +187,7 @@ public function filterBy($filterName, $value, $operator = false, DriverInterface
...
                 $filter->find('css', 'button.dropdown-toggle')->click();
-                $filter->find('css', '[data-value="'.$operator.'"]')->click();
+                $filter->find('css', '[data-value="' . $operator . '"]')->click();
```
Comment (phaseinducer) [inline features/Context/Page/Base/Grid.php]: `sprintf()` would be nice here.

Comment (sumbobyboys) [inline features/Context/Page/Base/Grid.php]: :+1:

**Pull request merged**